packages/x-date-pickers/src/internals/hooks/usePicker/usePickerViews.ts

```diff
@@ -185,7 +184,7 @@ export const usePickerViews = <
   TAdditionalProps
 >): UsePickerViewsResponse<TView> => {
   const { onChange, open, onClose } = propsFromPickerValue;
-  const { views, openTo, onViewChange, disableOpenPicker, viewRenderers, timezone } = props;
+  const { views, openTo, onViewChange, viewRenderers, timezone } = props;
```

Comment:
You may be able to simplify the typing and get rid of `UsePickerViewsNonStaticProps` if `disableOpenPicker` is no longer used in this file

Reply:
Not exactly
From what I see, `disableOpenPicker` is defined in `UsePickerViewsNonStaticProps`
Which is weird now that this prop is never used in the `usePickerViews` hook.
If you move it to `UseDesktopPickerProps`, it would be removed from the mobile pickers (which make sense since it does nothing on them).

Or maybe I'm missing something, I did not dive deeper tbh

Reply:
On Mobile, it currently doesn't change anything, but on range pickers, it prevents the picker from opening on click. 🤔 
That's why I didn't move anything more than needed now.

Reply:
OK
The current position of this prop's definition seems odd to me but we can move it later if we find a better interface to put it in.